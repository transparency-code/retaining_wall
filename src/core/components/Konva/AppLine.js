/**
 * AppPoint Class
 * 
 */
import { AppPoint } from './AppPoint';

//Line have startpoint, endpoint and ID
//functions
export class AppLine {

  constructor(startPoint, endPoint, lineID) {
    if (!(startPoint instanceof AppPoint) || !(endPoint instanceof AppPoint)) {
      throw new Error(
        `Line Constructor . X and Y should be points. Received ${startPoint.toString()} and ${endPoint.toString()}`
      );
    }

    this.startPoint = startPoint;
    this.endPoint = endPoint;
    this.lineID = lineID;
  }

  getPointArray() {
    const points = [this.startPoint.x, this.startPoint.y, this.endPoint.x, this.endPoint.y];
    return points;
  }

  getAnEndPointForLine(
    directionx,
    directiony,
    movex,
    movey
  ) {
    return this.endPoint.createSuccesivePoint(directionx, directiony, movex, movey);
  }

  createSuccesiveLine(
    directionx,
    directiony, 
    movex,
    movey,
    lineID,
  ) {
    const newline = new AppLine(this.endPoint, this.getAnEndPointForLine(directionx, directiony, movex, movey));

    if (lineID) {
      newline.lineID = lineID;
    }

    return newline;
  }

  getXOffsetPointfromlineEnd(xOffset) {
    const newPoint = new AppPoint(this.endPoint.x + xOffset, this.endPoint.y);
    return newPoint;
  }

  gettagpointXforLabel() {
    return (this.startPoint.x + this.endPoint.x) / 2;
  }

  gettagpointYforLabel() {
    return (this.startPoint.y + this.endPoint.y) / 2;
  }
}