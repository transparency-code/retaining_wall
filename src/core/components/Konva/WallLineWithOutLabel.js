import React from 'react'
import {  Line } from 'react-konva';
import { lineColor , backgroundColor} from './constants';

/**
 * 
 * @inv{bool} true if line is invisible
 */
export function WallLineWithOutLabel(props) {

    const {line,inv} = props;
    
   let color = lineColor

    if (inv) {
      color = backgroundColor
    }
  
    // console.log(color)
    return <Line points={line.getPointArray()} stroke={color} />;
  }