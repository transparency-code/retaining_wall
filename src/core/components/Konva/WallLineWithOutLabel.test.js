import React from 'react';
import renderer from 'react-test-renderer';
import { WallLineWithOutLabel } from './WallLineWithOutLabel'

import { AppLine } from './AppLine'
import { exportAllDeclaration } from '@babel/types';
//ES6 Class Mocks
jest.mock('./AppLine'); // Appline is now a mock constructor


//https://www.youtube.com/watch?v=XDkSaCgR8g4


beforeEach(() => {
    // Clear all instances and calls to constructor and all methods:
    AppLine.mockClear();
});

it('check if the consumer called the class constructor', () => {
    const appLine = new AppLine()
    expect(AppLine).toHaveBeenCalledTimes(1);
})

it('check if the consumer called a method on the class instance', () => {

    const appLine = new AppLine()
    // https://jestjs.io/docs/en/mock-function-api.html#mockfnmockinstances
    // An array that contains all the object instances that have been instantiated from this mock function using new
    const getPointArray = AppLine.mock.instances[0].getPointArray
    appLine.getPointArray()
    expect(getPointArray).toHaveBeenCalledTimes(1);
})

it('snapshoting a visible line', () => {

    const appLine = new AppLine()
    const tree = renderer
        .create(<WallLineWithOutLabel line={appLine} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('snapshoting an invisible line', () => {

    const appLine = new AppLine()
    const tree = renderer
        .create(<WallLineWithOutLabel line={appLine} inv={true} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});