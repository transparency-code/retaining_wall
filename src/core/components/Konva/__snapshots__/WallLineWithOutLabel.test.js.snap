// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshoting a visible line 1`] = `
<Line
  stroke="grey"
/>
`;

exports[`snapshoting an invisible line 1`] = `
<Line
  stroke="white"
/>
`;
